### ISSUE TITLE
Provide option to process whole directory/jar/package for `extract-annotation` 

### ISSUE THREAD

**Issue by the1derer** [typetools/annotation-tools#200]
## **Problem**

The main problem for this arises from the fact that at present state we have to use `extract-annotations` for every file separately or have to write multiple Class-Names.

This puts the burden on the user to enter the name of every class while processing a directory or a package.

## **Solution, Usage and Desired Result**

The idea is to create two additional flags for `extract-annotations` i.e. `--all-classes-in-folder` and `--all-classes-in-package`.

#### (a) `--all-classes-in-folder` 

**Usage:-**`extract-annotations --all-classes-in-folder /path/to/folder`

**Example:-** `extract-annotations --all-classes-in-folder  java/lang`

**Desired Result**
Using this flag `extract-annotation` will search for every CLASS file in _/path/to/folder_ directory and its subdirectories and generate a corresponding JAIF file in separate directory *JAIF/folder*
having same directory structure as *folder* but contacting JAIF files in place of CLASS files and no other file.

#### (b) `--all-classes-in-package` will  search the package/jar:-

Usage (i):-`extract-annotations --recursively-search-package path.to.package`   
Usage (ii):-`extract-annotations --all-classes-in-package JAR_NAME`

Example (i):- `extract-annotations --recursively-search-package java.lang`  
Example (ii):- `extract-annotations --all-classes-in-package checker-qual.jar`  

**Desired Result**
A directory named *JAIF/JAR_NAME* parallel to *JAR_NAME.jar* file or package having same directory structure as JAR or package but contating JAIF files in place of CLASS files.

**Comment by mernst** [comment]
Could you please create a branch and write the user documentation in that branch?

**Comment by the1derer** [comment]
> Could you please create a branch and write the user documentation in that branch?

Mike,
I tried using `extract-annotation *.class` and it solved all the issues that my proposal tries to solve. So I don't think these features make any sense now.

Thank you very much for guiding me through this, I learned a lot on how to proceed while trying to add/propose a new feature. This will surely help me while I propose Bazel's Developer to add Nullness Checker to their source code.
